source/docs/appendices/migration-guides.md

```diff
@@ -42,6 +42,504 @@ In this release we added pagination for orders. In the process, the `orders` Gra
+ */
+
+/**
+- * @param {import('graphql').DocumentNode} Query
+- * @param {import('@apollo/react-hooks/lib/types').LazyQueryHookOptions} options
+- * @returns {import('@apollo/react-hooks/lib/types').QueryTuple<any,import('@apollo/react-common').OperationVariables>}
++ * @param {UseLazyQueryParams[0]} Query
++ * @param {UseLazyQueryParams[1]} options
++ * @returns {UseLazyQueryReturn}
+ */
+
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/modules/Wishlist/AddAllWishlistToCartModal/useAddAllWishlistToCartModalInternal.js`
+
+```diff
+-import { useMutation } from "@apollo/react-hooks";
++import { useMutation } from "react-apollo";
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/modules/Wishlist/WishlistProvider/WishlistProvider.js`
+
+```diff
+-import { useLazyQuery, useQuery } from "@apollo/react-hooks";
++import { useLazyQuery, useQuery } from "react-apollo";
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/pages/Account/DownloadableProduct/EnhanceDownloadableProduct.js`
+
+```diff
+-         * @param {import('@apollo/react-hoc/lib/types').OptionProps<any, GraphQLData, any>} param0
++         * @param {import('react-apollo').OptionProps<any, GraphQLData, any>} param0
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/pages/Orders/Orders.js`
+
+```diff
+-import { useLazyQuery } from "@apollo/react-hooks";
++import { useLazyQuery } from "react-apollo";
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/pages/Product/useProductBySkuLoader.js`
+
+```diff
+-import { useQuery } from "@apollo/react-hooks";
++import { useQuery } from "react-apollo";
+```
+
+</td>
+</tr>
+
+</table>
+
+</details>
+
+- If after the above fixes there is still `@apollo/*` dependencies and you are not sure how to resolve please contact us for support. You can also check [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=8fae8438d46200cbd310b16c30b3078da28b7d42) and [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=418a291f917660b18672b6be2eb2778cb51dfef9) for the changes we did to our codebase.
+
+- Resolvers for fields not in the `schema` now throw errors. To avoid this remove all unneeded `field`/`Type` resolvers that do not have a representation in the GraphQL `schema`. Check our [cleaning commit](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1216/diffs?commit_id=1e2e04e7f28dba2ebd36da50701478a13a58a238) regarding this issue.
+- `makeApolloClientStub` now expects resolvers as an argument instead of mocks. This in in tandem with the [`graphql-tools` upgrade notes](https://www.graphql-tools.com/docs/mocking#migration-from-v7-and-below). Note that we provided an adapter that converts `mocks` to `resolvers` so that old tests are not affected. However it is recommended to make the change. To disable the adapter's warning set `FRONT_COMMERCE_WEB_PRINT_MAKEAPOLLOCLIENTSTUB_MOCKS_DEPRECATION_WARNING` environment variable to `"false"`.
```

Comment:
```suggestion
- `makeApolloClientStub` now expects resolvers as an argument instead of mocks. This is in tandem with the [`graphql-tools` upgrade notes](https://www.graphql-tools.com/docs/mocking#migration-from-v7-and-below). Note that we provided an adapter that converts `mocks` to `resolvers` so that old tests are not affected. However it is recommended to make the change. To disable the adapter's warning set `FRONT_COMMERCE_WEB_PRINT_MAKEAPOLLOCLIENTSTUB_MOCKS_DEPRECATION_WARNING` environment variable to `"false"`.
```